File: frontend/src/types/tool.ts
export interface ToolParameter {
  type: string;
  properties: Record<string, { type: string; description?: string }>;
  required?: string[];
}

export interface ToolDefinition {
  id: string;
  name: string;
  display_name: string;
  description: string;
  category: string;
  risk_level: 'low' | 'medium' | 'high';
  parameters: ToolParameter;
  is_enabled: boolean;
  created_at?: string;
  updated_at?: string;
}

export interface ToolCreateData {
  name: string;
  display_name: string;
  description?: string;
  category?: string;
  risk_level?: string;
  parameters?: Record<string, unknown>;
  is_enabled?: boolean;
}

export interface ToolUpdateData {
  display_name?: string;
  description?: string;
  category?: string;
  risk_level?: string;
  parameters?: Record<string, unknown>;
  is_enabled?: boolean;
}

export interface ToolFilters {
  category?: string;
  risk_level?: string;
  is_enabled?: boolean;
  page?: number;
  page_size?: number;
}

export interface ToolListResponse {
  items: ToolDefinition[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface ToolCallRecord {
  id: string;
  goal_id?: string | null;
  task_id?: string | null;
  agent_id?: string | null;
  worker_id?: string | null;
  tool_name: string;
  tool_input: Record<string, unknown>;
  tool_output?: string | null;
  status: 'started' | 'completed' | 'failed';
  latency_ms: number;
  error_message?: string | null;
  created_at?: string;
}

export interface ToolCallFilters {
  goal_id?: string;
  task_id?: string;
  agent_id?: string;
  tool_name?: string;
  page?: number;
  page_size?: number;
}

export interface ToolCallListResponse {
  items: ToolCallRecord[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: backend/src/schemas/tool.py
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field, ConfigDict


class ToolCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    display_name: str = Field(..., min_length=1, max_length=100)
    description: str = Field(default="")
    category: str = Field(default="其他", min_length=1, max_length=50)
    risk_level: str = Field(default="low", pattern=r"^(low|medium|high)$")
    parameters: Dict[str, Any] = Field(default_factory=dict)
    is_enabled: bool = True


class ToolUpdate(BaseModel):
    display_name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = None
    category: Optional[str] = Field(None, min_length=1, max_length=50)
    risk_level: Optional[str] = Field(None, pattern=r"^(low|medium|high)$")
    parameters: Optional[Dict[str, Any]] = None
    is_enabled: Optional[bool] = None


class ToolOut(BaseModel):
    id: str
    name: str
    display_name: str
    description: str
    category: str
    risk_level: str
    parameters: Dict[str, Any]
    is_enabled: bool
    created_at: Optional[str] = None
    updated_at: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ToolListResponse(BaseModel):
    items: List[ToolOut]
    total: int
    page: int
    page_size: int
    total_pages: int


class ToolCallRecordOut(BaseModel):
    id: str
    goal_id: Optional[str] = None
    task_id: Optional[str] = None
    agent_id: Optional[str] = None
    worker_id: Optional[str] = None
    tool_name: str
    tool_input: Dict[str, Any]
    tool_output: Optional[str] = None
    status: str
    latency_ms: int
    error_message: Optional[str] = None
    created_at: Optional[str] = None
    result: Dict[str, Any] = Field(default_factory=dict)

    model_config = ConfigDict(from_attributes=True)


class ToolCallListResponse(BaseModel):
    items: List[ToolCallRecordOut]
    total: int
    page: int
    page_size: int
    total_pages: int
